### ISSUE TITLE
Unable to execute a Direct Flow without specifying a globalSender

### ISSUE THREAD

**Issue by helgeskjellevik** [digipost/signature-api-client-java#169]
We're currently upgrading the signature-api-client-java from version 2.2. to version 4.6.

In our setup, we use a DirectClient with no global sender specified in the ClientConfiguration.
Instead we specify the sender’s organization number on each and every signature job.

In version 4.6, we encounter a problem with this method call, which throws an exception regardless if the job.getSender() is present or not.

no.digipost.signature.client.asice.CreateASiCE#createASiCE(JOB job):Line 40

``` java
        Sender sender = job.getSender()
                .orElse(globalSender
                .orElseThrow(SENDER_NOT_SPECIFIED));
```

The current workaround is to specify a dummy value for globalSender.

**Comment by runeflobakk** [comment]
That is an awkwardly embarrassing bug, indeed. And it must have [been there since the ability to specify sender per job was introduced 5 years ago](https://github.com/digipost/signature-api-client-java/commit/8b58aec5c614c08d4bff5daf44a123b292c829c4).

Simple fix though: replace the code snippet with:
```java
Sender sender = job.getSender()
                .orElseget(() -> globalSender
                .orElseThrow(SENDER_NOT_SPECIFIED));
```
to avoid invoking `.orElseThrow` on `globalSender` unless `job.getSender()` returns `Optional.empty()`.

If you want to submit a pull-request, feel free! I will, though, be able to fix it myself at least within a few hours from now. And release v4.6.1 with the fix.

Thanks for reporting!

**Comment by runeflobakk** [comment]
There is a fix in v4.6.1, available in Maven Central Repository. There is also a pending release of v5.0.1 which should soon also be available.
